ticket_migrate.py

```diff
@@ -0,0 +1,829 @@
+        data = {
+            'content_id': self.github_id,
+            'content_type': 'Issue'
+            }
+
+        if DRY_RUN:
+            print(f"Would post to {url}:")
+            pprint.pprint(data)
+            return
+
+        response = requests.post(
+            url=url,
+            headers={'accept': 'application/vnd.github.v3+json'},
+            json=data,
+            auth=(config.OAUTH_USER, config.OAUTH_TOKEN)
+            )
+
+        if response.status_code != 201:
+            print('Error: Issue not added to project column!')
+            print(response)
+            pprint.pprint(response.json())
+            import pdb
+            pdb.set_trace()
+
+    @classmethod
+    def fromTracData(
+            cls,
+            component,
+            owner,
+            summary,
+            description,
+            priority,
+            keywords,
+            **kwargs):
+        """
+        Create a GitHubRequest from Trac ticket data fields.
+        """
+        return cls(
+            owner=config.OWNER,
+            repo=get_repo(component),
+            trac_id=kwargs['t_id'],
+            title=summary,
+            body=get_body(description, data=kwargs),
+            closed=kwargs['status'] == 'closed',
+            milestone=None,
+            labels=get_labels(
+                component,
+                priority,
+                keywords,
+                kwargs['status'],
+                kwargs['resolution']
+                ),
+            assignees=get_assignees(owner),
+            )
+
+    @classmethod
+    def fromTracDataMultiple(cls, trac_data):
+        """
+        Generate GitHubRequests from an iterable of dicts of Trac tickets.
+        """
+        for ticket in trac_data:
+            yield cls.fromTracData(**ticket)
+
+    @staticmethod
+    def commentFromTracData(trac_data):
+        """
+        Convert Trac comment data to GitHub comment body as JSON.
+        """
+        author, _ = config.USER_MAPPING.get(
+            trac_data['author'],
+            (trac_data['author'], 'ignored-email-field')
+            )
+
+        body = (
+            f"Comment by {author} at {showtime(trac_data['c_time'])}.\n"
+            f"\n"
+            f"{parse_body(trac_data['newvalue'])}"
+            )
+
+        return {'body': body}
+
+    def _submitComment(self, comment_data):
+        """
+        Send a POST request to GitHub creating the comment from `comment_data`.
+        """
+        url = f'https://api.github.com/repos/' \
+              f'{self.owner}/{self.repo}/issues/{self.github_number}/comments'
+
+        if DRY_RUN:
+            print(f"Would post to {url}:\n{comment_data}")
+            return
+
+        if not self.github_number:
+            print(f"Trac ticket {self.t_id} did not receive a GitHub number.")
+            import pdb
+            pdb.set_trace()
+
+        # Avoid crossing the rate limit (5000/hr).
+        sleep(1)
```

Comment:
nice try :) 

Note that in the response of each header you should receive the remaining limit